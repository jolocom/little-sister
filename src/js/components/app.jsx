import React from 'react'
import Reflux from 'reflux'
import Radium from 'radium'
import {bankUri} from 'lib/fixtures'

import {Layout, Content} from 'components/layout'
import {
  Paper,
  AppBar,
  IconButton,
  Snackbar,
  FlatButton,
  Dialog} from 'material-ui'
import Badge from 'material-ui/Badge'
import NavigationMenu from 'material-ui/svg-icons/navigation/menu'

import getMuiTheme from 'material-ui/styles/getMuiTheme'

import JolocomTheme from 'styles/jolocom-theme'

import LeftNav from 'components/left-nav/nav.jsx'
import Profile from 'components/accounts/profile.jsx'
import Tour from 'components/tour.jsx'

import Loading from 'components/common/loading.jsx'

import AccountActions from 'actions/account'
import AccountStore from 'stores/account'
import ConfirmStore from 'stores/confirm'

import PinnedActions from 'actions/pinned'
import ConfirmActions from 'actions/confirm'
import ProfileActions from 'actions/profile'
import ProfileStore from 'stores/profile'

import SnackbarStore from 'stores/snackbar'

// A pathname is considered public if either "/" or if it starts
// with any of the following publicRoutes
const publicRoutes = [
  '/login',
  '/signup',
  '/forgot-password',
  '/change-password'
]

let App = React.createClass({

  mixins: [
    Reflux.connect(AccountStore, 'account'),
    Reflux.connect(ProfileStore, 'profile'),
    Reflux.connect(SnackbarStore, 'snackbar'),
    Reflux.connect(ConfirmStore, 'confirm')
  ],

  propTypes: {
    location: React.PropTypes.object,
    children: React.PropTypes.node,
    route: React.PropTypes.object
  },

  contextTypes: {
    router: React.PropTypes.object.isRequired
  },

  childContextTypes: {
    muiTheme: React.PropTypes.object,
    profile: React.PropTypes.any,
    account: React.PropTypes.object,
    username: React.PropTypes.string,
    searchActive: React.PropTypes.bool,
    location: React.PropTypes.object,
    route: React.PropTypes.object
  },

  getChildContext: function () {
    let {account, profile, searchActive} = this.state
    return {
      muiTheme: this.theme,
      profile,
      account,
      username: account && account.username, // backward compat
      searchActive,
      location: this.props.location,
      route: this.props.route
    }
  },

  getInitialState() {
    return {
      searchActive: false,
      searchQuery: null,
      showIndicator: false
    }
  },

  componentWillMount() {
    this.theme = getMuiTheme(JolocomTheme)
    AccountActions.login()
  },

  componentDidUpdate(prevProps, prevState) {
    let {username} = this.state.account

    if (prevState.account.username === undefined ||
      prevState.account.username !== username) {
      this.checkLogin()
    }
  },

  isPublicRoute(path = this.props.location.pathname) {
    return path === '/' ||
      publicRoutes.some((publicRoute) => path.indexOf(publicRoute) === 0)
  },

  checkLogin() {
    let {username, loggingIn} = this.state.account

    // session is still loading, so return for now
    if (username === undefined && loggingIn) {
      return
    }

    if (!username && !this.isPublicRoute()) {
      this.context.router.push('/login')
    } else if (username && this.isPublicRoute()) {
      this.context.router.push('/graph')
    }

    if (username) {
      ProfileActions.load()
    }
  },

  _handlePinnedTap() {
    PinnedActions.show()
  },

  _handleSearchTap() {
    this.refs.search.show()
    this.setState({searchActive: true})
  },

  _handleSearchChange(query) {
    this.setState({searchQuery: query})
  },

  _handleSearchSubmit() {
    let uri = `${bankUri}/${this.state.searchQuery}#this`
    this.context.router.push(`/graph/${encodeURIComponent(uri)}`)
    this.refs.search.hide()
  },

  _handleSearchHide() {
    this.setState({
      searchActive: false,
      searchQuery: null
    })
  },

  _handleChatTap() {
    this.context.router.push('/conversations')
  },

  showDrawer() {
    this.refs.leftNav.show()
  },

  _handleConfirmClose() {
    ConfirmActions.close()
    this.state.confirm.callback()
  },

  getStyles() {
    let styles = {
      container: {
        width: '100%',
        height: '100%',
        position: 'relative'
      },
      header: {
        zIndex: 5,
        backgroundColor: this.theme.appBar.color
      },
      bar: {
        boxShadow: 'none'
      },
      nav: {
        display: this.state.searchActive ? 'none' : 'block'
      },
      icon: {
        color: this.theme.appBar.textColor
      },
      filters: {
        width: '100%',
        height: '48px'
      },
      menuIcon: {
        marginTop: '-20px',
        position: 'relative',
        top: '-4px',
        cursor: 'pointer'
      },
      hamburgerBadge: {
        top: 10,
        right: 20,
        width: 15,
        height: 15,
        display: 'none'
      }
    }
    return styles
  },

  render() {
    const styles = this.getStyles()

    // @TODO render login screen when logging in, also makes sures child
    // components don't get rendered before any user data is available
    if (this.state.account.loggingIn && !this.isPublicRoute()) {
      return <Loading />
    }

    // Deactivating search until we get it working
    /* <IconButton
       iconClassName="material-icons"
       iconStyle={styles.icon}
       onTouchTap={this._handleSearchTap}>search</IconButton> */
    const nav = (
      <div>
        <IconButton
          iconClassName="material-icons"
          iconStyle={styles.icon}
          onTouchTap={this._handleChatTap}>chat</IconButton>
      </div>
    )

    // Deactivating search until we get it working
    /*
    (
      <GraphSearch
        ref="search"
        onChange={this._handleSearchChange}
        onSubmit={this._handleSearchSubmit}
        onHide={this._handleSearchHide} />
    )
    */
    const search = null

    // Deactivating the filters until we get them working
    // <GraphFilters style={styles.filters} showDefaults />
    const filters = null

    const confirmActions = [
      <FlatButton
        label="Cancel"
        primary
        onTouchTap={this._handleConfirmClose}
      />,
      <FlatButton
        label={this.state.confirm.primaryActionText}
        primary
        onTouchTap={this._handleConfirmClose}
      />
    ]

    return (
      <div style={styles.container}>
        {this.isPublicRoute() ? this.props.children : (
          <Layout>
            <Paper style={styles.header}>
              <AppBar
                title="Graph"
                iconElementRight={nav}
                style={styles.bar}
                iconElementLeft={
                  <Badge
                    badgeContent={''}
                    secondary
                    badgeStyle={styles.hamburgerBadge}>
                    <NavigationMenu
                      onTouchTap={this.showDrawer}
                      style={styles.menuIcon} />
                  </Badge>
                }
                onLeftIconButtonTouchTap={this.showDrawer} />
              {filters}
              {search}
            </Paper>
            <LeftNav ref="leftNav" />
            <Content>
              {React.Children.map(this.props.children, (el) => {
                return React.cloneElement(el, {
                  searchQuery: this.state.searchQuery
                })
              })}
            </Content>
            <Profile />
            <Tour />
          </Layout>
        )}

        <Snackbar
          open={this.state.snackbar.open}
          message={this.state.snackbar.message}
          action={this.state.snackbar.undo && 'undo'}
          onActionTouchTap={this.state.snackbar.undoCallback}
        />

        <Dialog
          actions={confirmActions}
          modal={false}
          open={this.state.confirm.open}
          onRequestClose={this.handleClose}
        >
          {this.state.confirm.message}
        </Dialog>
      </div>

    )
  }

})

export default Radium(App)
