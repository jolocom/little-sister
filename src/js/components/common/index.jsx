export { default as IconIdCard } from './icon-idcard'
export { default as IconPassport } from './icon-passport'
export { default as IconServices } from './icon-services'
export { default as Loading } from './loading'
export { default as Error } from './error'
export { default as Spinner } from './spinner'
export { default as IconCheckmark } from './icon-checkmark'
export { default as IconEther } from './icon-ether'
export { default as IconBlockchain } from './icon-blockchain'
