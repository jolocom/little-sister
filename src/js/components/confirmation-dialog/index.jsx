import React from 'react'
import { connect } from 'redux/utils'

import { FlatButton, Dialog } from 'material-ui'

@connect({
  props: ['confirm'],
  actions: ['confirmation-dialog:close']
})
export default class ConfirmationDialog extends React.Component {
  static propTypes = {
    close: React.PropTypes.func,
    confirm: React.PropTypes.object
  }

  _handleConfirmAction() {
    this.props.close()
    this.props.confirm.callback() // Action when the user confirms
  }

  _handleConfirmCancel() {
    this.props.close()
  }

  render() {
    const confirmHandler = () => this._handleConfirmAction()
    const cancelHandler = () => this._handleConfirmCancel()

    const confirmActions = [
      <FlatButton
        label="Cancel"
        primary
        onTouchTap={cancelHandler}
      />,
      <FlatButton
        label={this.props.confirm.primaryActionText}
        primary
        onTouchTap={confirmHandler}
      />
    ]

    return <Dialog
      actions={confirmActions}
      modal={false}
      open={this.props.confirm.open}
      onRequestClose={this.handleClose}
    >
      {this.props.confirm.message}
    </Dialog>
  }
}
