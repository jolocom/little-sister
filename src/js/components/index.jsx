import React from 'react'
import Radium from 'radium'
import {RaisedButton} from 'material-ui'
import Carousel from 're-carousel'
import IndicatorDots from 'components/common/indicator-dots.jsx'

let Index = React.createClass({
  contextTypes: {
    router: React.PropTypes.object,
    username: React.PropTypes.string,
    muiTheme: React.PropTypes.object
  },

  componentWillMount() {
    const {account} = this.context
    if (account && account.webId) {
      this.context.router.push('/graph')
    }
  },

  getStyles() {
    let {muiTheme} = this.context

    let styles = {
      container: {
        width: '100%',
        height: '100%',
        position: 'relative',
        display: 'flex',
        flexDirection: 'column',
        backgroundColor: muiTheme.jolocom.gray4
      },
      logo: {
        fontSize: '18px',
        fontWeight: '400',
        textAlign: 'center',
        marginTop: '24px',
        textTransform: 'uppercase'
      },
      logoImg: {
        width: '32px',
        height: '32px',
        verticalAlign: 'middle'
      },
      intro: {
        flex: 1,
        textAlign: 'center'
      },
      slide: {
        backgroundColor: muiTheme.jolocom.gray4,
        padding: '24px',
        height: '100%',
        display: 'flex',
        flexDirection: 'column'
      },
      content: {
        flex: 1
      },
      img: {
        maxWidth: '100%'
      },
      title: {
        color: muiTheme.jolocom.gray1,
        fontSize: '28px',
        fontWeight: '300'
      },
      logoStartImg: {
        padding: '24px',
        maxWidth: '80%',
        width: '320px',
        margin: 'auto',
        marginTop: '20vh'
      },
      actions: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'stretch',
        padding: '24px',
        maxWidth: '80%',
        width: '260px',
        margin: 'auto'
      },
      signup: {
        marginBottom: '12px'
      },
      login: {
      }
    }

    return styles
  },

  render() {
    let styles = this.getStyles()

    return (
      <div style={styles.container}>

        {/** <div style={styles.logo}>
          <img src="/img/logo.png" style={styles.logoImg} /> Jolocom
        </div> **/}

        {/** <Carousel style={styles.intro} indicator={IndicatorDots}>
          <div style={styles.slide}>
            <h3 style={styles.title}>Own your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-data.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Create your personal digital identity</h3>
            <div style={styles.content}>
              <img src="/img/slide-identity.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Own your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-share.png" style={styles.img} />
            </div>
          </div>
          <div style={styles.slide}>
            <h3 style={styles.title}>Visualize your data</h3>
            <div style={styles.content}>
              <img src="/img/slide-visualize.png" style={styles.img} />
            </div>
          </div>
        </Carousel> **/}

        <div style={styles.intro}>
          <img
            src="/img/logo_littlesister_start.svg"
            style={styles.logoStartImg} />
        </div>

        <div style={styles.actions}>
          <RaisedButton
            secondary
            label="Sign up"
            style={styles.signup}
            onClick={this._handleSignup}
          />
          <RaisedButton
            label="Log in"
            style={styles.login}
            onClick={this._handleLogin}
          />
        </div>
      </div>
    )
  },

  _handleSignup() {
    this.context.router.push('/signup')
  },

  _handleLogin() {
    this.context.router.push('/login')
  }

})

export default Radium(Index)
