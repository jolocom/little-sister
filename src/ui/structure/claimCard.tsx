import React, { ReactNode } from 'react'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import {
  View,
  Text,
  StyleSheet,
  TextStyle,
  ViewStyle,
  RegisteredStyle,
} from 'react-native'

/**
 * Atomic UI elements are claim card wrapper and claim card
 * Section claim card combines the claim card outer wrapper and claim card
 */

interface ClaimCardWrapperProps {
  leftIcon: ReactNode
  rightIcon?: ReactNode
  containerStyle?: ViewStyle | RegisteredStyle<ViewStyle>
  title?: string
}

export const ClaimCardWrapper: React.SFC<ClaimCardWrapperProps> = props => {
  const styles = {
    defaultContainerStyle: {
      flexDirection: 'row',
      paddingTop: '3%',
      justifyContent: 'space-between',
      backgroundColor: JolocomTheme.primaryColorWhite,
    } as ViewStyle,
    defaultTitleStyle: {
      alignSelf: 'flex-start',
      ...JolocomTheme.textStyles.light.labelDisplayFieldEdit,
      color: '#05050d',
      fontFamily: JolocomTheme.contentFontFamily,
    } as ViewStyle,
    defaultClaimSectionStyle: {
      flex: 0.65,
      justifyContent: 'space-between',
    } as ViewStyle,
    defaultLeftIconStyle: {
      alignItems: 'center',
      flex: 0.2,
    } as ViewStyle,
    defaultRightIconStyle: {
      flex: 0.15,
    },
  }

  return (
    <View style={[styles.defaultContainerStyle, props.containerStyle]}>
      <View style={styles.defaultLeftIconStyle}>{props.leftIcon}</View>
      <View style={styles.defaultClaimSectionStyle}>
        <Text style={styles.defaultTitleStyle}>
          {props.title ? props.title : ''}
        </Text>
      </View>
      <View style={styles.defaultRightIconStyle}>
        {props.rightIcon || null}
      </View>
    </View>
  )
}

interface ClaimCardProps {
  rightIcon?: ReactNode
  secondaryText?: string | ReactNode
  primaryText: string | ReactNode
  primaryTextStyle?: TextStyle | RegisteredStyle<TextStyle>
  secondaryTextStyle?: TextStyle | RegisteredStyle<TextStyle>
  containerStyle?: ViewStyle | RegisteredStyle<ViewStyle>
}

export const ClaimCard: React.SFC<ClaimCardProps> = props => {
  const styles = StyleSheet.create({
    primaryTextDefault: {
      fontFamily: JolocomTheme.contentFontFamily,
      fontSize: JolocomTheme.headerFontSize,
      color: JolocomTheme.primaryColorBlack,
      fontWeight: '100',
    } as TextStyle,
    secondaryTextDefault: {
      fontSize: 17,
      opacity: 0.4,
    } as TextStyle,
    containerDefault: {
      flex: 0.8,
      marginBottom: '5%',
      justifyContent: 'center',
    },
  })

  const {
    primaryText,
    secondaryText,
    rightIcon,
    primaryTextStyle,
    secondaryTextStyle,
    containerStyle,
  } = props
  const { primaryTextDefault, secondaryTextDefault, containerDefault } = styles

  return (
    <View
      style={{
        display: 'flex',
        flexDirection: 'row',
        backgroundColor: 'white',
      }}
    >
      <View style={[containerDefault, containerStyle]}>
        <Text style={[primaryTextDefault, primaryTextStyle]}>
          {primaryText}
        </Text>
        <Text
          style={[primaryTextDefault, secondaryTextDefault, secondaryTextStyle]}
        >
          {secondaryText}
        </Text>
      </View>
      <View flex={0.2}>{rightIcon}</View>
    </View>
  )
}

interface SectionClaimCardProps {
  title: string
  primaryText: string | ReactNode
  leftIcon: ReactNode
  rightIcon?: ReactNode
  secondaryText?: string | ReactNode
  primaryTextStyle?: TextStyle | RegisteredStyle<TextStyle>
  secondaryTextStyle?: TextStyle | RegisteredStyle<TextStyle>
  containerStyleWrapper?: ViewStyle | RegisteredStyle<ViewStyle>
  containerStyleClaimCard?: ViewStyle | RegisteredStyle<ViewStyle>
}

export const SectionClaimCard: React.SFC<SectionClaimCardProps> = props => {
  const styles = StyleSheet.create({
    mainContainer: {
      width: '100%',
    },
    containerStyleClaimCardDefault: {
      paddingLeft: '20%',
    },
    containerStyleWrapperDefault: {
      paddingTop: '5%',
      marginTop: '1%',
    },
  })

  return (
    <View style={styles.mainContainer}>
      <ClaimCardWrapper
        leftIcon={props.leftIcon}
        containerStyle={
          props.containerStyleWrapper || styles.containerStyleWrapperDefault
        }
        title={props.title}
      />
      <ClaimCard
        primaryText={props.primaryText}
        secondaryText={props.secondaryText}
        primaryTextStyle={props.primaryTextStyle}
        secondaryTextStyle={props.secondaryTextStyle}
        containerStyle={
          props.containerStyleClaimCard || styles.containerStyleClaimCardDefault
        }
        rightIcon={props.rightIcon}
      />
    </View>
  )
}
