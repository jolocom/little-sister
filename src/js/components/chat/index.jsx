import React from 'react'
import Radium from 'radium'

import {AppBar, Tabs, Tab, Paper, IconButton} from 'material-ui'

import Dialog from 'components/common/dialog.jsx'
import {Layout, Content} from 'components/layout'

import GraphStore from 'stores/graph-store'

@Radium
export default class Chat extends React.Component {
  static contextTypes = {
    router: React.PropTypes.any,
    store: React.PropTypes.object
  }

  static propTypes = {
    children: React.PropTypes.node,
    location: React.PropTypes.object
  }

  constructor(props) {
    super(props)

    this.state = {
      activeTab: this.getActiveTab(props.location.pathname)
    }
  }

  getActiveTab(path) {
    let activeTab = 'chat'

    if (path === '/contacts') {
      activeTab = 'contacts'
    }

    return activeTab
  }

  componentWillUpdate(newProps) {
    if (this.props.location.pathname !== newProps.location.pathname) {
      this.setState({
        activeTab: this.getActiveTab(newProps.location.pathname)
      })
    }
  }

  close() {
    if (GraphStore.state.center == null) {
      this.context.router.push('/graph')
    } else {
      this.context.router.push('/graph/' +
        encodeURIComponent(GraphStore.state.center.uri))
    }
  }

  render() {
    const backIcon = (
      <IconButton
        iconClassName="material-icons"
        iconStyle={styles.icon}
        onTouchTap={this._handleBackTouchTap}
      >
        arrow_back
      </IconButton>
    )

    /*
    const searchIcon = (
      <IconButton
        iconClassName="material-icons"
        iconStyle={styles.icon}
      >
        search
      </IconButton>
    )
    */

    return (
      <Dialog id="chat" visible fullscreen>
        <Layout>
          <Paper>
            <AppBar
              title="Chat"
              style={styles.bar}
              iconElementLeft={backIcon}
              // iconElementRight={searchIcon}
            />
            <Tabs valueLink={{
              value: this.state.activeTab,
              requestChange: (tab) => this._handleTabsChange(tab)
            }}>
              <Tab label="Conversations" value="chat" />
              <Tab label="Contacts" value="contacts" />
            </Tabs>
          </Paper>
          <Content style={styles.conversationsContainer}>
            {this.props.children}
          </Content>
        </Layout>
      </Dialog>
    )
  }

  _handleTabsChange(tab) {
    this.setState({activeTab: tab})

    switch (tab) {
      case 'chat':
        this.context.router.push('/conversations')
        break
      case 'contacts':
        this.context.router.push('/contacts')
        break
    }
  }

  _handleBackTouchTap = () => {
    this.close()
  }
}

let styles = {
  bar: {
    boxShadow: 'none'
  },
  conversationsContainer: {
    overflowY: 'auto'
  }
}
