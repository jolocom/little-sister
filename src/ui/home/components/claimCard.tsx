import React from 'react'
import { ListItem } from 'react-native-material-ui'
import { StyleSheet, View } from 'react-native'
import { JolocomTheme } from 'src/styles/jolocom-theme'
import { IClaimUI } from 'src/ui/home/components/claimOverview'
import {
  MoreIcon,
  AccessibilityIcon,
  NameIcon,
  EmailIcon,
  PhoneIcon
} from 'src/resources'

interface Props {
  openClaimDetails: (selectedType: string) => void
  claimItem: IClaimUI
}

interface IIconMap {
  [key: string]: any
}

const iconMap : IIconMap = {
  name: <NameIcon />,
  email: <EmailIcon />,
  telephone: <PhoneIcon />
}

// TODO FIX THIS
const styles = StyleSheet.create({
  containerField: {
    width: '100%',
    marginBottom: 1
  },
  listItemPrimaryTextContainer: {
    height: 21,
    top: -3
  },
  listItemLeftElementContainer: {
    height: 24,
    width: 24,
    marginTop: 20,
    marginLeft: 16,
    marginRight: 30,
    marginBottom: 35
  },
  listItemRightElementContainer: {
    height: 24,
    width: 12,
    marginTop: 16,
    marginRight: 16,
    marginBottom: 35
  },
  listItemContainerOneLine: {
    height: 79
  },
  listItemContainerTwoLines: {
    paddingLeft: 54,
    marginLeft: 0,
    height: 79
  }
})

export const ClaimCard : React.SFC<Props> = (props) => {
  const { claimValue, claimField } = props.claimItem
  let content = []

  if (claimField === 'name' && claimValue !== undefined) {
    const splitName = claimValue.split(' ')
    content.push({
      claimValue: splitName[0],
      claimField,
      label: 'firstName',
      showIcon: true
    }, {
      claimValue: splitName[1],
      claimField,
      label: 'lastName',
      showIcon: false
    })
  } else {
    content.push({
      claimValue,
      claimField,
      label: claimField,
      showIcon: true
    })
  }

  const renderLeftIcon = (claimField: string, showIcon: boolean) => {
    if (!showIcon) {
      return ''
    } else if (iconMap[claimField] !== undefined) {
      return iconMap[claimField]
    } else if (iconMap[claimField] === undefined) {
      return <AccessibilityIcon />
    }
  }

  const renderCard = (claimVal: any, claimField: string, label: string, showIcon: boolean) => {
    const {
      labelDisplayFieldEdit,
      labelDisplayField,
      textDisplayFieldEdit,
      textDisplayField
    } = JolocomTheme.textStyles.light

    return (
      <ListItem
        key={ label }
        style={{
          primaryTextContainer: styles.listItemPrimaryTextContainer,
          primaryText: claimVal === undefined ? labelDisplayFieldEdit : labelDisplayField,
          secondaryText: {
            ...claimVal === undefined ? textDisplayFieldEdit : textDisplayField,
            lineHeight: 24
          },
          container: showIcon ? styles.listItemContainerOneLine : styles.listItemContainerTwoLines,
          leftElementContainer: styles.listItemLeftElementContainer,
          rightElementContainer: styles.listItemRightElementContainer
        }}
        centerElement={{
          primaryText: stringCapitalize(label),
          secondaryText: claimVal === undefined ?
          '+ add' :
          claimVal
        }}
        leftElement={ renderLeftIcon(claimField, showIcon) }
        onPress={() => {}}
        rightElement={ claimVal !== undefined ? <MoreIcon /> : '' }
      />
    )
  }

  return(
    <View style={ styles.containerField }>
      { content.map((c) => {
        return renderCard(c.claimValue, c.claimField, c.label, c.showIcon)
      }) }
    </View>
  )
}

const stringCapitalize = (myString : string) : string => {
  const matches = myString.match(/[A-Z]/g)
  if (matches !== null) {
    matches.map((match) => {
      const index = myString.indexOf(match)
      const tx = myString.slice(0, index) + " " + myString.slice(index)
      myString = tx
    })
  }
  return myString[0].toUpperCase() + myString.slice(1)
}
