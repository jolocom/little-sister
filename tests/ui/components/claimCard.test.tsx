import React from 'react'
import { ClaimCard } from 'src/ui/home/components/claimCard'
import { shallow } from 'enzyme'

describe('ClaimCard component', ()=> {
  it('matches the snapshot on render', () => {
    const props = {
      claimType: 'name',
      firstClaimLabel: 'name',
      openClaimsDetails: () => null
    }
    const rendered = shallow(<ClaimCard
      {...props}
    />)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot of a two line claim', () => {
    const props = {
      claimType: 'name',
      firstClaimLabel: 'first name',
      firstClaimValue: 'natascha',
      claimLines: 2,
      secondClaimLabel: 'last name',
      secondClaimValue: 'xx',
      openClaimsDetails: () => null
    }
    const rendered = shallow(<ClaimCard
      {...props}
    />)
    expect(rendered).toMatchSnapshot()
  })

  it('matches the snapshot of a claim with unspecified icon / uses default icon', () => {
    const props = {
      claimType: 'favColor',
      firstClaimLabel: 'favourite Color',
      firstClaimValue: 'green',
      openClaimsDetails: () => null
    }

    const rendered = shallow(<ClaimCard
      {...props}
    />)
    expect(rendered).toMatchSnapshot()
  })
})
