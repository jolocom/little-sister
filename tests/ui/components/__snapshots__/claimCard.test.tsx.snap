// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClaimCard component matches the snapshot of a claim with unspecified icon / uses default icon 1`] = `
<Component
  style={
    Object {
      "marginBottom": 1,
      "width": "100%",
    }
  }
>
  <ListItem
    centerElement={
      Object {
        "primaryText": "Fav Color",
        "secondaryText": "green",
      }
    }
    dense={false}
    divider={false}
    key="favColor"
    leftElement={<AccessibilityIcon />}
    numberOfLines={1}
    onLeftElementPress={null}
    onLongPress={null}
    onPress={[Function]}
    onPressValue={null}
    onRightElementPress={null}
    rightElement={<MoreIcon />}
    style={
      Object {
        "container": Object {
          "height": 79,
        },
        "leftElementContainer": Object {
          "height": 24,
          "marginBottom": 35,
          "marginLeft": 16,
          "marginRight": 30,
          "marginTop": 20,
          "width": 24,
        },
        "primaryText": Object {
          "color": "black",
          "fontSize": 17,
          "opacity": 0.4,
        },
        "primaryTextContainer": Object {
          "height": 21,
          "top": -3,
        },
        "rightElementContainer": Object {
          "height": 24,
          "marginBottom": 35,
          "marginRight": 16,
          "marginTop": 16,
          "width": 12,
        },
        "secondaryText": Object {
          "color": "black",
          "fontSize": 22,
          "lineHeight": 24,
        },
      }
    }
    testID={null}
  />
</Component>
`;

exports[`ClaimCard component matches the snapshot of a two line claim 1`] = `
<Component
  style={
    Object {
      "marginBottom": 1,
      "width": "100%",
    }
  }
>
  <ListItem
    centerElement={
      Object {
        "primaryText": "First Name",
        "secondaryText": "natascha",
      }
    }
    dense={false}
    divider={false}
    key="firstName"
    leftElement={<NameIcon />}
    numberOfLines={1}
    onLeftElementPress={null}
    onLongPress={null}
    onPress={[Function]}
    onPressValue={null}
    onRightElementPress={null}
    rightElement={<MoreIcon />}
    style={
      Object {
        "container": Object {
          "height": 79,
        },
        "leftElementContainer": Object {
          "height": 24,
          "marginBottom": 35,
          "marginLeft": 16,
          "marginRight": 30,
          "marginTop": 20,
          "width": 24,
        },
        "primaryText": Object {
          "color": "black",
          "fontSize": 17,
          "opacity": 0.4,
        },
        "primaryTextContainer": Object {
          "height": 21,
          "top": -3,
        },
        "rightElementContainer": Object {
          "height": 24,
          "marginBottom": 35,
          "marginRight": 16,
          "marginTop": 16,
          "width": 12,
        },
        "secondaryText": Object {
          "color": "black",
          "fontSize": 22,
          "lineHeight": 24,
        },
      }
    }
    testID={null}
  />
  <ListItem
    centerElement={
      Object {
        "primaryText": "Last Name",
        "secondaryText": "world",
      }
    }
    dense={false}
    divider={false}
    key="lastName"
    leftElement=""
    numberOfLines={1}
    onLeftElementPress={null}
    onLongPress={null}
    onPress={[Function]}
    onPressValue={null}
    onRightElementPress={null}
    rightElement={<MoreIcon />}
    style={
      Object {
        "container": Object {
          "height": 79,
          "marginLeft": 0,
          "paddingLeft": 54,
        },
        "leftElementContainer": Object {
          "height": 24,
          "marginBottom": 35,
          "marginLeft": 16,
          "marginRight": 30,
          "marginTop": 20,
          "width": 24,
        },
        "primaryText": Object {
          "color": "black",
          "fontSize": 17,
          "opacity": 0.4,
        },
        "primaryTextContainer": Object {
          "height": 21,
          "top": -3,
        },
        "rightElementContainer": Object {
          "height": 24,
          "marginBottom": 35,
          "marginRight": 16,
          "marginTop": 16,
          "width": 12,
        },
        "secondaryText": Object {
          "color": "black",
          "fontSize": 22,
          "lineHeight": 24,
        },
      }
    }
    testID={null}
  />
</Component>
`;

exports[`ClaimCard component matches the snapshot on render 1`] = `
<Component
  style={
    Object {
      "marginBottom": 1,
      "width": "100%",
    }
  }
>
  <ListItem
    centerElement={
      Object {
        "primaryText": "Name",
        "secondaryText": "+ add",
      }
    }
    dense={false}
    divider={false}
    key="name"
    leftElement={<NameIcon />}
    numberOfLines={1}
    onLeftElementPress={null}
    onLongPress={null}
    onPress={[Function]}
    onPressValue={null}
    onRightElementPress={null}
    rightElement=""
    style={
      Object {
        "container": Object {
          "height": 79,
        },
        "leftElementContainer": Object {
          "height": 24,
          "marginBottom": 35,
          "marginLeft": 16,
          "marginRight": 30,
          "marginTop": 20,
          "width": 24,
        },
        "primaryText": Object {
          "color": "black",
          "fontSize": 17,
        },
        "primaryTextContainer": Object {
          "height": 21,
          "top": -3,
        },
        "rightElementContainer": Object {
          "height": 24,
          "marginBottom": 35,
          "marginRight": 16,
          "marginTop": 16,
          "width": 12,
        },
        "secondaryText": Object {
          "color": "#942f51",
          "fontSize": 22,
          "lineHeight": 24,
        },
      }
    }
    testID={null}
  />
</Component>
`;
